warehouse/tuf/services.py

```diff
@@ -0,0 +1,408 @@
+        fileinfo["hashes"] = hashes
+        if custom:
+            fileinfo["custom"] = custom
+
+        return fileinfo
+
+    def _set_expiration_for_role(self, role_name):
+        # If we're initializing TUF for development purposes, give
+        # every role a long expiration time so that developers don't have to
+        # continually re-initialize it.
+        if self._request.registry.settings["warehouse.env"] == Environment.development:
+            return datetime.datetime.now().replace(microsecond=0) + datetime.timedelta(
+                seconds=self._request.registry.settings[
+                    "tuf.development_metadata_expiry"
+                ]
+            )
+        else:
+            return datetime.datetime.now().replace(microsecond=0) + datetime.timedelta(
+                seconds=self._request.registry.settings[f"tuf.{role_name}.expiry"]
+            )
+
+    def init_repository(self):
+        metadata_repository = MetadataRepository(
+            self._storage_backend, self._key_storage_backend
+        )
+
+        if metadata_repository.is_initialized:
+            raise FileExistsError("TUF Metadata Repository files already exists.")
+
+        top_roles_payload = dict()
+        for role in TOP_LEVEL_ROLE_NAMES:
+            top_roles_payload[role] = RolesPayload(
+                expiration=self._set_expiration_for_role(role),
+                threshold=self._request.registry.settings[f"tuf.{role}.threshold"],
+                keys=self._key_storage_backend.get(role, "private"),
+            )
+
+        metadata_repository.initialize(top_roles_payload, True)
+
+    def bump_snapshot(self):
+        """
+        Bump Snapshot Role Metadata
+        """
+        # Bumping the Snapshot role involves the following steps:
+        # 1. Initiate Metadata Repository.
+        # 2. Load the Snapshot Role.
+        # 3. Bump Snapshot role (and write to the Storage).
+        # 4. Bump Timestamp role using the new Snapshot Metadata.
+        # 1. Metadata Repository.
+        metadata_repository = MetadataRepository(
+            self._storage_backend, self._key_storage_backend
+        )
+
+        # 2. Snapshot role metadata.
+        snapshot_metadata = metadata_repository.load_role(Role.SNAPSHOT.value)
+
+        # 3. Bump Snapshot role metadata.
+        snapshot_metadata = metadata_repository.snapshot_bump_version(
+            snapshot_expires=self._set_expiration_for_role(
+                self._request, Role.SNAPSHOT.value
+            ),
+            snapshot_metadata=snapshot_metadata,
+            store=True,
+        )
+
+        # 4. Bump Snapshot role metadata using Timestamp metadata.
+        metadata_repository.timestamp_bump_version(
+            snapshot_version=snapshot_metadata.signed.version,
+            timestamp_expires=self._set_expiration_for_role(
+                self._request, Role.TIMESTAMP.value
+            ),
+            store=True,
+        )
+
+    def bump_bins_ns(self):
```

Comment:
```suggestion
    def bump_bin_n_roles(self):
```

See comment about bin/bins/bin-n/bin-ns/... name disambiguation in cli/tuf.py (line 89).